### ISSUE TITLE
Replace file() with open() in docs/core

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8529]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8529|
|Type|enhancement|
|Created|2016-06-28 06:18:41Z|
|Branch|https://github.com/twisted/twisted/tree/8529-rodrigc-file|

In Python 3, file() is gone (see https://docs.python.org/3/whatsnew/3.0.html#builtins ), and needs to be replaced with open()
which works on Python 2 and 3

<details><summary>Searchable metadata</summary>

```
trac-id__8529 8529
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8529_rodrigc_file 8529-rodrigc-file
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1467094721850180 1467094721850180
changetime__1467116003102973 1467116003102973
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/251

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_2"></a>|@Lukasa set owner to @rodrigc|
|-|-|

Looks good to me, merge when ready.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub set status to `closed`|
|-|-|

In changeset eea4c6c41009252d96177479a230453798b3f738
```
#!CommitTicketReference repository="" revision="eea4c6c41009252d96177479a230453798b3f738"
Merge 8529-rodrigc-file: Replace file() with open() in docs/core

Author: rodrigc
Reviewer: Lukasa
Fixes: #8529
```
